diff --git a/Smarty/templates/Header.tpl b/Smarty/templates/Header.tpl
old mode 100644
new mode 100755
index 9cf754a..f9e10e0
--- a/Smarty/templates/Header.tpl
+++ b/Smarty/templates/Header.tpl
@@ -38,8 +38,23 @@
        <![endif]-->
 {/literal}
        <!-- End -->
+{if $MODULE neq 'FieldFormulas' && $MODULE neq 'com_vtiger_workflow' && $MODULE neq 'evvtApps'} 
+    
+<script src="Smarty/angular/angular.min.js"></script>
+<script  src="Smarty/angular/ng-table.js"></script>
+<script src="Smarty/angular/angular-multi-select.js"></script>  
+<link rel="stylesheet" href="Smarty/angular/angular-multi-select.css">
+<link data-require="ng-table@*" data-semver="0.3.0" rel="stylesheet" href="http://bazalt-cms.com/assets/ng-table/0.3.0/ng-table.css" />
+<script src="http://angular-ui.github.io/bootstrap/ui-bootstrap-tpls-0.6.0.js"></script>
+<link href="//netdna.bootstrapcdn.com/twitter-bootstrap/2.3.1/css/bootstrap-combined.min.css" rel="stylesheet">
+
+<script>
+angular.module('cbApp', ['ngTable','ui.bootstrap','multi-select']);
+</script>
+
+{/if}
 </head>
-       <body leftmargin=0 topmargin=0 marginheight=0 marginwidth=0 class=small>
+       <body leftmargin=0 topmargin=0 marginheight=0 marginwidth=0 class=small ng-app="cbApp">
        <a name="top"></a>
        <!-- header -->
        <!-- header-vtiger crm name & RSS -->

diff --git a/Smarty/templates/DetailView.tpl b/Smarty/templates/DetailView.tpl
index d1048ae..68f85ef 100755
--- a/Smarty/templates/DetailView.tpl
+++ b/Smarty/templates/DetailView.tpl

            {if preg_match("/^block:\/\/.*/", $CUSTOM_LINK_DETAILVIEWWIDGET->linkurl)}
-             {if  ($smarty.foreach.BLOCKS.first && $CUSTOM_LINK_DETAILVIEWWIDGET->sequence <= 1) 
+             {if  (is_numeric($CUSTOM_LINK_DETAILVIEWWIDGET->linklabel))
+                   || ($smarty.foreach.BLOCKS.first && $CUSTOM_LINK_DETAILVIEWWIDGET->sequence <= 1) 
                    || ($CUSTOM_LINK_DETAILVIEWWIDGET->sequence == $smarty.foreach.BLOCKS.iteration)

diff --git a/include/utils/VtlibUtils.php b/include/utils/VtlibUtils.php
old mode 100644
new mode 100755
index 12deae7..133dcfc
--- a/include/utils/VtlibUtils.php
+++ b/include/utils/VtlibUtils.php
@@ -599,7 +599,7 @@ function vtlib_process_widget($widgetLinkInfo, $context = false) {
                }
                if (class_exists($widgetControllerClass)) {
                        $widgetControllerInstance = new $widgetControllerClass;
-                       $widgetInstance = $widgetControllerInstance->getWidget($widgetLinkInfo->linklabel);
+                       $widgetInstance = $widgetControllerInstance->getWidget($widgetLinkInfo->linklabel,$context);
                        if ($widgetInstance) {
                                return $widgetInstance->process($context);
                        }

Smarty/angular     folder
Smarty/templates/modules/NgBlock      folder
modules/NgBlock            folder
Vtlib_NgBlock_extension.php
create_ngBlock_table.php